File: ielts_admin_dashboard/fix.py

import os

old_block = '''async function tryGetAdminDb() {
  try {
    const { getAdminDb } = await import('@/lib/firebase-admin');
    return getAdminDb();
  } catch {
    return null;
  }
}'''

new_block = '''async function tryGetAdminDb() {
  try {
    return getAdminDb();
  } catch (err: any) {
    console.error('Failed to init admin:', err);
    return null;
  }
}'''

for root, _, files in os.walk('src/app/api'):
    for f in files:
        if f.endswith('route.ts'):
            path = os.path.join(root, f)
            with open(path, 'r', encoding='utf-8') as file:
                content = file.read()
            
            if 'import { getAdminDb }' not in content:
                content = content.replace(
                    "import { NextResponse } from 'next/server';",
                    "import { NextResponse } from 'next/server';\nimport { getAdminDb } from '@/lib/firebase-admin';"
                )
            
            content = content.replace(old_block, new_block)
            
            with open(path, 'w', encoding='utf-8') as file:
                file.write(content)
print('Done!')

File: ielts_admin_dashboard/fix.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    file = dir + '/' + file;
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) { 
      results = results.concat(walk(file));
    } else { 
      if (file.endsWith('route.ts')) results.push(file);
    }
  });
  return results;
}

const files = walk('src/app/api');
files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  
  if (!content.includes('import { getAdminDb }')) {
    content = content.replace(
      'import { NextResponse } from \'next/server\';',
      'import { NextResponse } from \'next/server\';\nimport { getAdminDb } from \'@/lib/firebase-admin\';'
    );
  }

  const oldBlock = \sync function tryGetAdminDb() {
  try {
    const { getAdminDb } = await import('@/lib/firebase-admin');
    return getAdminDb();
  } catch {
    return null;
  }
}\;
  
  const newBlock = \sync function tryGetAdminDb() {
  try {
    return getAdminDb();
  } catch (err: any) {
    console.error('Failed to init admin:', err);
    return null;
  }
}\;
  
  content = content.replace(oldBlock, newBlock);
  fs.writeFileSync(file, content);
});
console.log('Replaced correctly.');
